src/fides/api/ctl/sql_models.py

```diff
@@ -579,3 +579,15 @@ class CustomField(Base):
     )
 
     UniqueConstraint("resource_type", "resource_id", "custom_field_definition_id")
+
+
+class AuditLogResource(Base):
+    """The log of user actions against fides resources."""
+
+    __tablename__ = "audit_log_resource"
+
+    user_id = Column(String, nullable=True, index=True)
```

Comment:
can this really be nullable? looking at the code, it's null in the case where we don't find an authorization header - but is that ever really the case? i guess it doesn't hurt too much to keep it flexible, so i'm not opposed to it. just wanted to flag it in case this wasn't a deliberate choice.

Reply:
Ya so it is null on login and null for the root user (I think I handled those so far) so after finding it in a couple of places I felt like keeping it as flexible as possible. If it fails I also want it to not really impact the user either (although I don't really know of a great way to be alerted to issues)

Reply:
that makes sense, let's keep it flexible for now!